lib/chef/transport.rb

```diff
@@ -0,0 +1,140 @@
+# Author:: Bryan McLellan <btm@loftninjas.org>
+# Copyright:: Copyright 2018, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef-config/mixin/credentials"
+require "train"
+
+class Chef
+  class Transport
```

Comment:
since this looks like a pile of helper class methods and you don't create an instance of this, it should probably be a module instead of a class (and then the methods no longer being class methods).

since it is train-specific it might be better named Chef::TrainTransport as well?

i was gonna suggest injecting the methods directly into the object but i think I like the inherent message to "go look in another file" with writing `Chef::TrainTransport.build_transport` rather than having `include Chef::TrainTransport .....  @transport = build_train_transport` which gives the user no clue of where to actually find build_train_transport (and you'd have to rename it because `build_transport` would then also be too generic...)